src/dbnode/persist/fs/seek.go

```diff
@@ -461,6 +461,76 @@ func (s *seeker) SeekIndexEntry(
 	}
 }
 
+// SeekIndexEntryToIndexHash performs the following steps:
+//
+//     1. Go to the indexLookup and it will give us an offset that is a good starting
+//        point for scanning the index file.
+//     2. Reset an offsetFileReader with the index fd and an offset (so that calls to Read() will
+//        begin at the offset provided by the offset lookup).
+//     3. Reset a decoder with fileDecoderStream (offsetFileReader wrapped in a bufio.Reader).
+//     4. Called DecodeIndexEntry in a tight loop (which will advance our position in the
+//        offsetFileReader internally) until we've either found the entry we're looking for or gone so
+//        far we know it does not exist.
+func (s *seeker) SeekIndexEntryToIndexHash(
```

Comment:
Seems that this method does the same as `SeekIndexEntry` just with less fields?

It would be better I believe to use the same underlying code with a flag/option to avoid returning the fields you don't want.

But also, with a streaming approach I think this becomes entirely different since you can avoid using the index summary lookup and shouldn't need to call "getNearestIndexFileOffset" at all when reading all the entries in an entire file (i.e. dense read).